neutrino.go

```diff
@@ -1727,6 +1783,70 @@ func (s *ChainService) Start() error {
 		return nil
 	}
 
+	// Preload headers before starting any subsystem. Block headers should be
+	// sideloaded first as the amount of filter headers to be sideloaded
+	//// depends on it.
+	if s.blkHdrSideloader != nil {
+		log.Debugf("Sideloading block headers")
+		err := s.blkHdrSideloader.sideLoadHeaders(s.blkHdrSideloader.
+			blkHdrCurHeight,
+			s.blkHdrSideloader.blkHdrProcessor.nextCheckpt.Height, true)
+		if err != nil {
+			return err
+		}
+
+	}
+
+	if s.cfHdrSideloader != nil {
+		_, blkTip, err := s.BlockHeaders.ChainTip()
+		if err != nil {
+			return err
+		}
+
+		s.cfHdrSideloader.fHdrLastHeight = blkTip
+		var doNotSideloadCfheaders bool
+		if s.cfHdrSideloader.fHdrSkipVerify {
```

Comment:
See my comment about a `HeaderValidator` and `HeaderWriter` for the cf headers. That way we can just call something like `sideLoadHeaders` instead of having all this logic be in line.

Reply:
Nice. I think I have that `sideLoadHeaders` function already: https://github.com/lightninglabs/neutrino/blob/a8b442494364badb0dea84a57975169e71b325e9/sideload.go#L362-L420
I put this here because after sideloading the block headers we want to _guide_ sideloading the cfheaders as the cfheaders tip should not surpass the block header's. Maybe I can abstract this logic into  a new function? Not sure how the `HeaderValidator` and `HeaderWriter` plays into this.

Reply:
Do you see how having the logic in line here defeats the purpose of making these crafted abstractions via interfaces? We want to hide as much information as possible. Instead, here we leak the details of checkpoints, etc. This routine should be concerned only with initial initialiaztion. 

It might be helpful to think top-down here, starting with an API sketch like this:
```go
if cfg.BlockHeaderLoad != nil {
    blkSideLoader := NewSideLoader(&BlockHeaderWriter{}, &BlockHeaderValidator{})
    if blkSideLoader.LoadHeaders(cfg.BlockHedaderLoad.Reader); err != nil {
         return err
    }
}

if cfg.FilterHeaderLoad != nil {
    filterSideLoader := NewSideLoader(&FilterHeaderWriter{}, &FilterHeaderValidator{})
    if FilerSideLoader.LoadHeaders(cfg.BlockHedaderLoad.Reader); err != nil {
         return err
    }
}
```

Reply:
Notice how that doesn't leak any unnecessary details into this context. Encapsulation is key.